.editorconfig

```diff
@@ -72,3 +72,9 @@ charset = utf-8-bom
 [*.{cs,vb}]
 # CA1305: Specify IFormatProvider
 dotnet_diagnostic.CA1305.severity = error
+
+[*.{cs,vb}]
+# CA2012: Use ValueTask correctly
+dotnet_diagnostic.CA2012.severity = warning
+[**/{test,perf}/**.{cs,vb}]
+dotnet_diagnostic.CA2012.severity = suggestion
```

Comment:
In dotnet/runtime we have a .ruleset for all src and a separate .ruleset for all tests.  I couldn't find anything like that in dotnet/aspnetcore, so I tried to achieve the same this way.  Please let me know if there's a better way to go about it. (I don't have any objections to the rule also being enabled for tests, there's just a ton of violations there at present and I couldn't tackle it as part of this change.)

Reply:
Sounds reasonable, I don't have any particular insight on it, however @dotnet/aspnet-build may have some thoughts/ideas

Reply:
Seems reasonable to me

Reply:
We have a couple of .editorconfig and .globalconfig files elsewhere in the repo e.g. src/Http/.editorconfig and src/Mvc/MvcTests.globalconfig but @Youssef1313 did a good job avoiding those "except if" cases. If the wildcard is enough to avoid creating new files, I'm all for it.